data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourceCoordinator.java

```diff
@@ -97,6 +97,18 @@ public interface SourceCoordinator<T> {
      */
     void giveUpPartitions();
 
+    /**
+     * Should be called by the source when it is shutting down to indicate that it will no longer be able to perform work on partitions,
+     * or can be called to give up ownership of its partitions in order to pick up new ones with {@link #getNextPartition(Function)} ()}.
+     * Same as giveUpPartitions() but with specific partition to be
+     * given up. This can be used when source coordinator is shared
+     * by multiple threads.
+     *
+     * @throws org.opensearch.dataprepper.model.source.coordinator.exceptions.PartitionUpdateException if the partition could not be given up due to some failure
+     * @since 2.8
+     */
+    void giveUpPartitions(SourcePartition<T> sourcePartition);
```

Comment:
Let's name this `giveUpPartition` instead of `giveUpPartitions`

Reply:
Is it a bit redundant with closePartition?

Reply:
closePartition does not make partitions available (i.e. UNASSIGNED)